packages/vite/src/node/external.ts

```diff
@@ -150,7 +151,9 @@ function createIsExternal(
     }
     let isExternal = false
     if (id[0] !== '.' && !path.isAbsolute(id)) {
-      isExternal = isBuiltin(id) || isConfiguredAsExternal(id, importer)
+      isExternal =
+        isBuiltin(id, environment.config) ||
```

Comment:
This should be `environment.config.resolve.isBuiltin(id)` to me, and the default value can be always false for the client as you proposed and the current node-targeted function for consumer server.

Reply:
Done 😄👍